_posts/2022-07-19-react-to-full-stack.md

```diff
@@ -0,0 +1,606 @@
+
+You should see a login page, however, we haven't set up authentication as yet - so it won't work. Which takes us to the next stage, configuring Cognito for authentication.
+
+<br>
+
+### Creating a Cognito user pool for authentication
+
+[AWS Cognito](https://aws.amazon.com/cognito/) lets you add user sign-up, sign-in, and access control to your web and mobile apps quickly and easily. It really does make handling authentication a breeze.
+
+Before all of these steps, please ensure your AWS region is correct. If you do not know which region to pick, check [this](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/) out and pick the one closest to you.
+
+- To do this simply click the drop down and select your region. I am based in Perth, Western Australia so my closest region is ap-southeast-2.
+
+  ![](/img/react-to-full-stack/region.png)
+
+1. From the AWS console search for Cognito.
+
+   ![](/img/react-to-full-stack/cognito-1.png)
+
+2. Click 'Manage User Pools'.
+
+   ![](/img/react-to-full-stack/cognito-2.png)
+
+3. In the top right corner click 'Create a user pool'.
+
+   ![](/img/react-to-full-stack/cognito-3.png)
+
+4. Enter a name for your user pool then click 'Review Defaults'.
+
+   ![](/img/react-to-full-stack/cognito-4.png)
+
+5. Click 'Add app client'.
+
+   ![](/img/react-to-full-stack/cognito-5.png)
+
+6. Click 'Add an app client'.
+
+   ![](/img/react-to-full-stack/cognito-6.png)
+
+7. Create a name for your app client, uncheck "Generate client secret", then scroll down and click 'Create app client'.
+
+   ![](/img/react-to-full-stack/cognito-7.png)
+
+8. Click 'Return to pool details'.
+
+   ![](/img/react-to-full-stack/cognito-8.png)
+
+9. Click 'Create pool'.
+
+   ![](/img/react-to-full-stack/cognito-9.png)
+
+10. Copy your 'Pool Id' and paste it in userPoolId:"" in the COGNITO_CONFIG object in constants.ts (todo-app > src >constants > constants.ts) of the todo-app repo.
+
+    ![](/img/react-to-full-stack/cognito-10.png)
+
+    Your region is stated before the underscore, in my case it is ap-southeast-2. Plug in both the region and userPoolId into the COGNITO_CONFIG object. At this point it should look something like this.
+
+    ```javascript
+    export const COGNITO_CONFIG = {
+      region: "ap-southeast-2",
+      userPoolId: "ap-southeast-2_mmrMEg55o",
+      userPoolWebClientId: "",
+    };
+    ```
+
+11. Navigate to 'App clients' using the side panel.
+
+    ![](/img/react-to-full-stack/cognito-11.png)
+
+12. Copy the 'App client Id' and paste it into the 'userPoolWebClientId' value of the COGNITO_CONFIG object.
+
+    ![](/img/react-to-full-stack/cognito-12.png)
+
+    At this point your COGNITO_CONFIG object is configured and should look something like this.
+
+    ```javascript
+    export const COGNITO_CONFIG = {
+      region: "ap-southeast-2",
+      userPoolId: "ap-southeast-2_mmrMEg55o",
+      userPoolWebClientId: "6gjhe86c07bg688qhtvfqf8mdi",
```

Comment:
hm.. I'm not sure, but are you exploiting your demo app by providing your readers with this config??

Reply:
If you are, maybe "mask" the values 😅

Reply:
The demo todo-app has empty object values, this is just used as an example to show them what their config should look like.